### PR TITLE
implemented the home page, with some errors

### PR DESCRIPTION
**Pull request by mihaimazareanu**
I keep getting "Hydration failed" error message, plus some other minor issues...

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,9 +1,96 @@
...
+        <FlexContainer>
+          <StyledDiv>
+            <p>
+              Don&apos;t take our word for it though, please register and try it
+              out now.
+            </p>
+            <RegisterButton
+              register={register}
+              handleClickRegister={handleClickRegister}
+            />
```
Comment (actyralf) [inline pages/index.js]: Please follow the convention of naming the props `on...` and only the handler `handle...`. So this line should read `onClickRegister={handleClickRegister}` (you have to adjust the prop name in your component as well)

File: pages/index.js [hunk 2]
```diff
@@ -1,9 +1,96 @@
...
+                fill="#5E8C49"
+                stroke="#5E8C49"
+                strokeWidth="0.9"
+              />
+            </svg>
+          </StyledDiv>
+        </FlexContainer>
+        {register && (
+          <RegisterForm loginMode={loginMode} setLoginMode={setLoginMode} />
+        )}
```
Comment (actyralf) [inline pages/index.js:71]: See comment above: prop names for callbacks should start with `on...`(e.g. `onChangeLoginMode`). Furthermore, you should never pass the state-setter directly to your component. Instead, write a function `handleChangeLoginMode` that accepts the new `loginMode` and handles the state change accordingly.

File: pages/profile.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import styled from "styled-components";
+
+export default function Profile() {
+  return (
+    <>
+      <Head>
+        <title>kidsFi - Finance for kids</title>
+      </Head>
+      <Body>
+        <p>Under Construction</p>
```
Comment (actyralf) [inline pages/profile.js:11]: Here you still have the `<Body>` component which might lead to the hydration errors.

**Review by actyralf — CHANGES_REQUESTED** [review]
Well done!

Please consider the comments on coding conventions and remove the remaining `<Body>` component from `/pages/profile.js`

File: components/RegisterForm.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+  showPassword,
+  onShowPassword,
+}) {
+  const [inputPassword, setInputPassword] = useState("");
+  const [inputRepeatPassword, setInputRepeatPassword] = useState("");
+
+  const handleChangePassword = event => {
+    setInputPassword(event.target.value);
+    console.log(inputPassword);
+  };
```
Comment (actyralf) [inline components/RegisterForm.js]: Please remove all `console.log` before creating a pull request. A production version should not produce debug output.

File: pages/_app.js [hunk 5]
```diff
@@ -3,12 +3,12 @@ import Layout from "../components/Layout";
 
 function MyApp({Component, pageProps}) {
   return (
-    <>
+    <main>
       <GlobalStyles />
       <Layout>
```
Comment (actyralf) [inline pages/_app.js]: Please move `<GobalStyles />` up one level (outside of main). It should affect everything.

**Review by actyralf — CHANGES_REQUESTED** [review]
Great work!

Two very small changes (see below)

File: pages/_app.js (hunk 5 above)
Comment (mihaimazareanu) [reply pages/_app.js]: Done. 😄

**Review by actyralf — CHANGES_REQUESTED** [review]
I think you forgot to remove the `console.log` debug messages...

**Comment by mihaimazareanu** [comment]
I didn't forget it, I didn't know if I should "touch" anything what was in the template in the beginning, so I left the octopodes.js document as it was, including the `console.log(error)` message. 

I took it out now. I hope all is good now?

**Review by actyralf — APPROVED** [review]
👍

**Pull request merged**